File: ci/tests/fixtures/sec_auth_03_ast_only.py
# SEC-AUTH-03 fixture with SPLIT expectations, same idea as the .js one.
#
#   semgrep MUST stay silent. Measured against PyJWT 2.13.0: an algorithms list of none
#   raises InvalidAlgorithmError rather than accepting a tampered token, so on Python this
#   is not a bypass and reporting it would be crying wolf.
#
#   the shell hook IS expected to fire. It greps a line, and `algorithms=["none"]` is the
#   same text in a .py file as in a .js file, where the same option genuinely does accept an
#   attacker-minted token once the key is empty. One line cannot carry which library it is.
#
# So the two tools disagree here on purpose, and the disagreement is the point: the hook
# warns and never blocks, semgrep is the one wired into the gate.
import jwt


def none_algorithm_is_refused_by_pyjwt(token, key):
    return jwt.decode(token, key, algorithms=["none"])

File: ci/tests/fixtures/sec_auth_03_ast_only.js
// SEC-AUTH-03 fixture with SPLIT expectations, and it exists to make a limit visible
// rather than to hide it.
//
//   semgrep MUST stay silent here. It parses, so it can see that the receiver is not a JWT
//   library and that the option belongs to something else entirely.
//
//   the shell hook IS expected to fire here. It greps a line, so `algorithms: ["none"]` is
//   the same text whatever object it belongs to. That is the price of a check that needs no
//   toolchain and runs on every write, and it is why the hook only ever warns while semgrep
//   is the one wired into the gate.
//
// If a future change makes the hook silent here, that is an improvement, and this file will
// fail until somebody moves it and says so on purpose.
function unrelatedVerifyWithTheSameOptionName(blob, key) {
  return archive.verify(blob, key, { algorithms: ["none"] });
}

function unrelatedCacheOption(cache, key) {
  return cache.get(key, { ignoreExpiration: true });
}
